src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -117,11 +117,11 @@ public static string GetFullPath(string path, string basePath)
                 combinedPath = JoinInternal(basePath, path);
             }
 
-            // Device paths are normalized by definition, so passing something of this format
-            // to GetFullPath() won't do anything by design. Additionally, GetFullPathName() in
-            // Windows doesn't root them properly. As such we need to manually remove segments.
+            // Device paths are normalized by definition, so passing something of this format (i.e. \\?\C:\.\tmp, \\.\C:\foo)
+            // to Windows APIs won't do anything by design. Additionally, GetFullPathName() in Windows doesn't root
+            // them properly. As such we need to manually remove segments and not use GetFullPath(). Note that we need
+            // to subtract one from GetRootLength in order to normalize the first segment after the root. eg "\\?\C:\..\", "\\?\C:\.\", "\\?\C:\\"
```

Comment:
This still doesn't explain why subtracting one works. You need to show the because.